swift/ios-shoppe-demo/Views/CheckoutTextViewCell.swift

```diff
@@ -30,22 +32,30 @@ class CheckoutTableViewCell: UITableViewCell, UITextFieldDelegate {
      # showTooltip sets the placeholder text on the textfield based on the provided enum case's placeholder property.
      */
     func showTooltip() {
-        if tableViewController?.autoFillEnabled ?? false { }
+        if checkoutTableViewController?.autoFillEnabled ?? false { }
         else {
             addressDetailTextField.placeholder = (addressDetail != nil) ? addressDetail?.placeHolder : cardDetail?.placeHolder
         }
 
         addressDetailTextField.delegate = self
     }
 
+    func setPlaceholder(_ name: String) {
+        addressDetailTextField.placeholder = name
+    }
+
     // MARK: - UITextFieldDelegate Methods
 
     func textFieldDidEndEditing(_ textField: UITextField) {
         if let addressDetail = addressDetail {
-            tableViewController?.addressDict[addressDetail] = addressDetailTextField.text
+            checkoutTableViewController?.addressDict[addressDetail] = addressDetailTextField.text
+            checkoutTableViewController?.tableView.reloadData()
+        }
+        else {
```

Comment:
It's usually preferred in Swift to have both braces on the same line as `else`.
```suggestion
        } else {
```